include/uinputdevice.h

```diff
@@ -33,13 +33,18 @@ class UinputDevice
   bool ready() const; 
   operator bool() const; 
   void destroy();
-private:
+protected:
   int _fd;
+  void open(std::string const& path);
 };
 
+void UinputDevice::open(std::string const& path) {
+  _fd = ::open(path.data(), O_WRONLY | O_NONBLOCK);
```

Comment:
This should probably first call `destroy()` to close any pre-existing file descriptors.